trin-core/src/portalnet/find/query_tracer.rs

```diff
@@ -0,0 +1,144 @@
+use crate::portalnet::Enr;
+
+use serde::Serialize;
+use std::collections::HashMap;
+use std::time::{Duration, SystemTime};
+
+///Store ENRs as strings for serialization simplicity
+type ENRString = String;
```

Comment:
If I remember right, `HashMap` couldn't use an Enr as a key.

Regardless, I've switched to using `NodeId`s as keys, and `NodeIdString` because `NodeId` doesn't implement `serde::Serialize`